core/geometry/src/curve/internalContexts/PolygonOffsetContext.ts

```diff
@@ -544,61 +588,89 @@ class Joint {
+  private static createOffsetSegment(
+    basePointA: Point3d, basePointB: Point3d, distance: number
+  ): CurvePrimitive | undefined {
     Vector3d.createStartEnd(basePointA, basePointB, this._unitAlong);
     if (this._unitAlong.normalizeInPlace()) {
       this._unitAlong.rotate90CCWXY(this._unitPerp);
       const segment = LineSegment3d.create(
         basePointA.plusScaled(this._unitPerp, distance, this._offsetA),
-        basePointB.plusScaled(this._unitPerp, distance, this._offsetB));
+        basePointB.plusScaled(this._unitPerp, distance, this._offsetB)
+      );
       CurveChainWireOffsetContext.applyBasePoints(segment, basePointA.clone(), basePointB.clone());
       return segment;
     }
     return undefined;
   }
-
   /**
-   * Construct curves that are offset from a polygon.
-   * * The construction will remove "some" local effects of features smaller than the offset distance, but will not detect self intersection with far-away edges.
-   * @param points
-   * @param wrap
-   * @param offsetDistance
+   * Construct a wire (not area) that is offset from given polyline or polygon (which must be in xy-plane or in
+   *  a plane parallel to xy-plane).
+   * * This is a simple wire offset (in the form of a line string), not an area.
+   * * If offsetDistance is given as a number, default OffsetOptions are applied.
+   * * When the offset needs to do an "outside" turn, the first applicable construction is applied:
+   *   * If the turn angle is larger than `options.minArcDegrees`, a circular arc is constructed.
+   *   * If the turn angle is less than or equal to `options.maxChamferTurnDegrees`, extend curves along tangent to
+   * single intersection point (to create a sharp corner).
+   *   * If the turn angle is larger than `options.maxChamferDegrees`, the turn is constructed as a sequence of
+   * straight lines (a line string) that are:
+   *      * outside the arc
+   *      * have uniform turn angle less than `options.maxChamferDegrees`
+   *      * each line segment (except first and last) touches the arc at its midpoint.
+   * * The construction algorithm attempts to eliminate some self-intersections within the offsets, but does not
+   * guarantee a simple area offset.
+   * @param points a single loop or path
+   * @param wrap true to include wraparound
+   * @param offsetDistanceOrOptions offset distance (positive to left of curve, negative to right) or JointOptions
+   * object.
    */
-  public constructPolygonWireXYOffset(points: Point3d[], wrap: boolean, leftOffsetDistanceOrOptions: number | JointOptions): CurveCollection | undefined {
+  public constructPolygonWireXYOffset(
+    points: Point3d[], wrap: boolean, leftOffsetDistanceOrOptions: number | JointOptions
+  ): CurveChain | undefined {
+    /**
+     * if "wrap = true", then first and last point in the points array must be close; otherwise
+     * generated offset will be invalid.
+     */
```

Comment:
If it's a short comment I prefer // but I prefer /* */ for longer comments and comments explaining all parts of the code (like this function) because it makes the code more readable (green color of comments vs gray) and since it's internal comment, has no effect outsideon users, I keep /** */

Reply:
IMHO, there's no need to take up 4 lines for this comment. In general, [long comments are discouraged.](https://www.itwinjs.org/learning/guidelines/typescript-coding-guidelines/#make-judicious-use-of-vertical-screen-space)